File: Behavioral/Mediator/Python/chat_mediator.py
# chat_mediator.py

from mediator import Mediator

class ChatMediator(Mediator):
  def __init__(self):
    self._users = []

  def add_user(self, user):
    self._users.append(user)

  def send_message(self, message, sender):
    for user in self._users:
      if user != sender:
        user.receive_message(message)

File: Behavioral/Mediator/JavaScript/chatMediator.js
// chatMediator.js

const Mediator = require("./mediator");

class ChatMediator extends Mediator {
  constructor() {
    super();
    this.users = [];
  }

  addUser(user) {
    this.users.push(user);
  }

  sendMessage(message, sender) {
    this.users.forEach(user => {
      if (user !== sender) {
        user.receiveMessage(message);
      }
    });
  }
}

module.exports = ChatMediator;
